### PR TITLE
feat!: Make React 18 as the only Peer Deps [JOB-63252] BREAKING CHANGE

### PR DESCRIPTION
**Pull request by darryltec**
## Motivations

There are a few more changes that need to happen that force us to have a major version bump that includes breaking changes.

### Added

- 

### Changed

- Change Toast from using ReactDOM `render` to `createRoot` [✅ DONE]
  - Reference: https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-client-rendering-apis
  - https://github.com/GetJobber/atlantis/pull/1114
- Bump @apollo/client to latest [✅ DONE]
  - Which also bump the peer deps
  - https://github.com/GetJobber/atlantis/pull/1105
- Bump react-router-dom to latest 5 [✅ DONE]
  - https://github.com/GetJobber/atlantis/pull/1103

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/package.json [hunk 1]
```diff
@@ -76,8 +76,8 @@
     "uuid": "^8.3.2"
   },
   "peerDependencies": {
-    "react": ">=16.8",
-    "react-dom": ">=16.8"
+    "react": "^18",
+    "react-dom": "^18"
```
Comment (timoteialbu) [inline packages/components/package.json:80]: Why is this needed? Should we not let consumers of this (ie non-jobber folk) decide on the react version? Some people do not check the dependencies for that like we do.

Comment (darryltec) [reply packages/components/package.json:80]: they want us to use this https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-client-rendering-apis but that only exists in R18.

**Review by kingstonfung — APPROVED** [review]

File: packages/components/src/Toast/showToast.tsx [hunk 2]
```diff
@@ -7,26 +7,27 @@ import React, {
...
-    target = document.createElement("div");
-    target.id = targetId;
-    target.classList.add(styles.wrapper);
-    document.body.appendChild(target);
-  }
+const root = createRoot(target);
 
-  render(<ToasterOven {...props} />, target);
+export function showToast(props: ToastProps) {
+  root.render(<ToasterOven {...props} />);
```
Comment (timoteialbu) [inline packages/components/src/Toast/showToast.tsx:30]: As discussed in Slack, we need this change to make the tests pass in JO.
```suggestion
  export function showToast(props: ToastProps) {
    if (document?.body?.contains(target)) {
      root.render(<ToasterOven {...props} />);
    }
}
```

**Review by timoteialbu — CHANGES_REQUESTED** [review]

**Comment by darryltec** [comment]
@timoteialbu pushed up a change in there! mind reviewing and giving a 👍🏼

**Review by kingstonfung — APPROVED** [review]

**Review by timoteialbu — APPROVED** [review]
